internal/types/option.go

```diff
@@ -0,0 +1,132 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/osbuild/images/internal/common"
+)
+
+// Option is a more constrained subset of the code in
+//     https://github.com/moznion/go-optional
+//
+// It is not using an external go-optional lib directly because none
+// has toml unmarshal support and also because there is no support for
+// complex types in UnmarshalTOML (it only gives a single
+// reflect.Value of type any) so our optional "lib" must come with
+// limitations for this.
+//
+// Unfortunately there is no way I could find to import the code and
+// limit the supported types. So this is a copy of the subset.
+// Fortunatly the code is small, targeted and easy to follow so it
+// should not be too bad.
+
+type OptionTomlTypes interface {
+	~int | ~bool | ~string
```

Comment:
Just for the record, you do not have to fix this here I think it would be better to have a common type in UBP:

```suggestion
	~bool | ~string | ~int | ~float64 | ~float32 | ~int8 | ~int16 | ~int32 | ~int64 | ~uint | ~uint8 | ~uint16 | ~uint32 | ~uint64
```